### PR TITLE
sendgrid-python: init at 6.1.1

### PR DESCRIPTION
**Pull request by xfoxawy**
###### Motivation for this change
Sendgrid is famous email service provider, my company depend on the python lib to send emails.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x ] NixOS
   - [ x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/python-http-client/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    sha256 = "1ncsvhb6sms2ahr737jmz1gq9ppn8yprj9v1awlnh8ylqg2scnq2";
+  };
+ 
+ propagatedBuildInputs = [ mock ];
+ 
+  meta = with lib; {
+    description = "Twilio SendGrid Quickly and easily access any RESTful or RESTful-like API";
+    homepage = https://pypi.org/project/python-http-client/;
+    license = licenses.mit;
+    maintainers = with maintainers; [ xfoxawy ];
```
Comment (kalbasit) [inline pkgs/development/python-modules/python-http-client/default.nix]: I don't see `xfoxawy` in the [maintainers/maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix). Please add it in a separate commit.

File: pkgs/development/python-modules/sendgrid/sendgrid/helpers/inbound/config.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Incoming Parse endpoint
```
Comment (kalbasit) [inline pkgs/development/python-modules/sendgrid/sendgrid/helpers/inbound/config.yml]: What is this file? I don't see it being used by the derivation.

**Review by kalbasit — CHANGES_REQUESTED** [review]
Thank you, @xfoxawy for the pull request!

The commit https://github.com/NixOS/nixpkgs/pull/78649/commits/45ba5227c2bacd9921556625cf274c0f58103973 does not follow the contributing guide. Please amend the commit and force push, with the other fixes.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -7522,6 +7522,11 @@
     githubId = 5121426;
     name = "Albert Safin";
   };
+  xfoxawy = {
+    email = "xfoxawy@gmail.com";
+    github = "xfoxawy";
+    name = "Mohammed Kamal";
+  }
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: please add githubId, forgot where the directions are to find this value, but should be easy to find.

File: pkgs/development/python-modules/sendgrid/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    sha256 = "5ad98598c13b8cf5545d20f6fb6ecf967a08b48e3e175affabd82cfc71522d01";
+  };
+
+  doCheck = false; # fails due to missing file sendgrid/helpers/inbound/config.yml, https://github.com/sendgrid/sendgrid-python/issues/868
+ 
+ propagatedBuildInputs = [ flask pyyaml six pytest mock codecov coverage python-http-client];
+ 
+  meta = with lib; {
+    description = "Twilio SendGrid library for Python";
+    homepage = https://pypi.org/project/sendgrid/;
```
Comment (jonringer) [inline pkgs/development/python-modules/sendgrid/default.nix]: This is just the pypi location
```suggestion
    homepage = "https://github.com/sendgrid/sendgrid-python";
```

File: pkgs/development/python-modules/sendgrid/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+buildPythonPackage rec {
+  pname = "sendgrid";
+  version = "6.1.1"; 
+ 
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "5ad98598c13b8cf5545d20f6fb6ecf967a08b48e3e175affabd82cfc71522d01";
+  };
+
+  doCheck = false; # fails due to missing file sendgrid/helpers/inbound/config.yml, https://github.com/sendgrid/sendgrid-python/issues/868
```
Comment (jonringer) [inline pkgs/development/python-modules/sendgrid/default.nix]: should be able to disable certain tests which care
```
  checkPhase = ''
    pytest --ignore=tests/test_app.py
  '';
```

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -7522,6 +7522,12 @@
     githubId = 5121426;
     name = "Albert Safin";
   };
+  xfoxawy = {
+    email = "xfoxawy@gmail.com";
+    github = "xfoxawy";
+    githubId = 5111856;
+    name = "Mohammed Kamal";
+  }
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:7530]: Please make this into a separate commit:

```
maintainers: add <name>
```

File: pkgs/development/python-modules/python-http-client/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1ncsvhb6sms2ahr737jmz1gq9ppn8yprj9v1awlnh8ylqg2scnq2";
+  };
+ 
+ propagatedBuildInputs = [ mock ];
+ 
+  meta = with lib; {
+    description = "Twilio SendGrid Quickly and easily access any RESTful or RESTful-like API";
+    homepage = https://pypi.org/project/python-http-client/;
```
Comment (jonringer) [inline pkgs/development/python-modules/python-http-client/default.nix:20]: rfc 45
```suggestion
    homepage = "https://pypi.org/project/python-http-client/";
```

File: pkgs/development/python-modules/sendgrid/default.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "5ad98598c13b8cf5545d20f6fb6ecf967a08b48e3e175affabd82cfc71522d01";
+  };
+
+ checkPhase = ''
+  test --ignore=tests/test_app.py
+ '';
+ 
+ propagatedBuildInputs = [ flask pyyaml six pytest mock codecov coverage python-http-client];
```
Comment (jonringer) [inline pkgs/development/python-modules/sendgrid/default.nix:28]: usual order for python packages
```suggestion
 propagatedBuildInputs = [ flask pyyaml six pytest mock codecov coverage python-http-client];
 
 checkPhase = ''
  test --ignore=tests/test_app.py
 '';
```

Comment (jonringer) [inline pkgs/development/python-modules/sendgrid/default.nix:28]: since this is quite a few packages, please do one per line
```suggestion
 checkPhase = ''
  test --ignore=tests/test_app.py
 '';
 
 propagatedBuildInputs = [
   flask 
   pyyaml
   ...
 ];
```

**Review by jonringer — COMMENTED** [review]
To comply with [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes) please have the commit message name be of the format
```
<pkg-name>: <subject-line>
```
for more examples, please look at https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes

in your case, the commit messages should be:
```
maintainers: add <name>
pythonPackages.python-http-client: init at <version>
pythonPackages.sendgrid: init at <version>
```

If you're unfamiliar editing git history, I would recommend looking into `git rebase --interactive` and `git add --patch`

**Comment by fabaff** [comment]
Superseded by #110456
